src/main/java/com/booklink/backend/service/impl/ForumServiceImpl.java

```diff
@@ -127,8 +125,13 @@ public Forum getForumEntityById(Long id) {
     @Override
     public List<ForumViewDto> searchForums(String searchTerm, Long userId) {
         if (searchTerm != null) {
-            List<Forum> forums = forumRepository.findDistinctByNameContainingIgnoreCaseOrTagsNameContainingIgnoreCase(searchTerm, searchTerm);
-            return ForumDtoFactory.createForumDtoAndForumViewDtoWithIsMember(forums, userId, ForumViewDto::from);
+            Set<Forum> forums = new HashSet<>(forumRepository.findDistinctByNameContainingIgnoreCaseOrTagsNameContainingIgnoreCase(searchTerm, searchTerm));
+            List<String> words = List.of(searchTerm.split("[ ,]"));
+
+            for (String word : words) {
+                forums.addAll(forumRepository.findAllByTagsNameContainingIgnoreCase(word));
+            }
```

Comment:
Aca en vez de `addAll` que funciona como un `and` necesitamos que funcione como un `or`.